pkg/artifactory/resource/repository/repository.go

```diff
@@ -278,6 +278,31 @@ func projectEnvironmentsDiff(_ context.Context, diff *schema.ResourceDiff, _ int
 func MkResourceSchema(skeema map[string]*schema.Schema, packer packer.PackFunc, unpack unpacker.UnpackFunc, constructor Constructor) *schema.Resource {
 	var reader = mkRepoRead(packer, constructor)
 	return &schema.Resource{
+		SchemaVersion: 2,
+		CreateContext: mkRepoCreate(unpack, reader),
+		ReadContext:   reader,
+		UpdateContext: mkRepoUpdate(unpack, reader),
+		DeleteContext: deleteRepo,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema:        skeema,
+		CustomizeDiff: projectEnvironmentsDiff,
+		StateUpgraders: []schema.StateUpgrader{
+			{
+				Type:    MkResourceSchemaV1(skeema, packer, unpack, constructor).CoreConfigSchema().ImpliedType(),
```

Comment:
`skeema` is the same for both v1 and v2. We need to define v1 and v2 schema separately (e.g. v1 has the additional attribute), and use them appropriately. There's no need for `MkResourceSchemaV1()` since all we need is 2 separate schemas.

Reply:
Added schema v1 and v2. Tested, and all works well.